hail/python/hailtop/hail_logging.py

```diff
@@ -63,6 +63,17 @@ def log(self, request, response, time):
             'x_real_ip': request.headers.get("X-Real-IP"),
         }
 
+        userdata_maybe = request.get('userdata', {})
+        if 'username' in userdata_maybe:
+            extra['username'] = userdata_maybe['username']
+        if 'login_id' in userdata_maybe:
+            extra['login_id'] = userdata_maybe['login_id']
+        if 'is_developer' in userdata_maybe:
+            extra['is_developer'] = userdata_maybe['is_developer']
+        if 'hail_identity' in userdata_maybe:
+            extra['hail_identity'] = userdata_maybe['hail_identity']
+
```

Comment:
I think this can be simplified to something more pythonic like (untested)
```python
id_keys = ['username', 'login_id', 'is_developer', 'hail_identity']
extra.update((k, userdata_maybe[k]) for k in id_keys if k in userdata_maybe)
```